IPython/core/tests/test_magic.py

```diff
@@ -623,9 +623,13 @@ def test_extension():
 @skipIf(dec.module_not_available('IPython.nbformat'), 'nbformat not importable')
 class NotebookExportMagicTests(TestCase):
     def test_notebook_export_json(self):
+        _ip = get_ipython()
+        _ip.history_manager.reset()   # Clear any existing history.
+        cmds = [u"a=1", u"def b():\n  return a**2", u"print(a, b())"]
```

Comment:
Fixed. I miss rebased so the test  of previous commit should pass, too lazy to split in commit in several ones;